### PR TITLE
lib/packagekit: Don't ever use PackageKit when /usr is mounted read-only

### PR DESCRIPTION
**Pull request by mvollmer**
PackageKit is sometimes installed on ostree systems, but that doesn't
mean we can install packages with it.

### CODE REVIEW

**Comment by mvollmer** [comment]
@garrett, this should make your storage page load properly on Silverblue.

File: pkg/lib/packagekit.js [hunk 1]
```diff
@@ -87,13 +87,19 @@ export function call(objectPath, iface, method, args, opts) {
...
+    return cockpit.spawn([ "findmnt", "-T", "/usr", "-n", "-o", "VFS-OPTIONS" ])
+            .then(options => {
+                if (options.split(",").indexOf("ro") >= 0)
+                    return false;
+                else
+                    return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
+                                "Get", [ "org.freedesktop.PackageKit", "VersionMajor" ])
+                            .then(() => true,
+                                  () => false);
+            });
```
Comment (martinpitt) [inline pkg/lib/packagekit.js]: Although the findmnnt Should Not Fail™, let's please add a catch() handler with a log at least.

File: pkg/storaged/client.js [hunk 2]
```diff
@@ -438,7 +438,7 @@ function init_model(callback) {
     }
 
     function enable_pk_features() {
-        return PK.detect().then(function (available) { client.features.packagekit = available });
+        return PK.detect().then(function (available) { console.log(available); client.features.packagekit = available });
```
Comment (martinpitt) [inline pkg/storaged/client.js]: Please either remove the log or add a text to it that says what the true/false value means.

Comment (mvollmer) [reply pkg/storaged/client.js]: Upps.

File: pkg/lib/packagekit.js (hunk 1 above)
Comment (mvollmer) [reply pkg/lib/packagekit.js]: I made it so that we fall back to checking the D-Bus API when findmnt fails.

Comment (mvollmer) [reply pkg/lib/packagekit.js]: Good catch!

File: pkg/lib/packagekit.js [hunk 3]
```diff
@@ -87,13 +87,24 @@ export function call(objectPath, iface, method, args, opts) {
...
+    }
+
+    return cockpit.spawn([ "findmnt", "-T", "/usr", "-n", "-o", "VFS-OPTIONS" ])
+            .then(options => {
+                if (options.split(",").indexOf("ro") >= 0)
+                    return false;
+                else
+                    return dbus_detect();
+            })
+            .catch(dbus_detect);
```
Comment (martinpitt) [inline pkg/lib/packagekit.js]: Thanks. Might still be interesting to console.warn() about this, so that we can spot it in the tests.

Comment (mvollmer) [reply pkg/lib/packagekit.js]: I think cockpit.js already does that: https://github.com/cockpit-project/cockpit/blob/master/src/base1/cockpit.js#L785

**Pull request merged**